crates/napi/src/next_api/endpoint.rs

```diff
@@ -90,9 +96,31 @@ impl Deref for ExternalEndpoint {
     }
 }
 
+// Await the source and return fatal issues if there are any, otherwise
+// propagate any actual error results.
+async fn strongly_consistent_catch_collectables<R: VcValueType + Send>(
+    source: Vc<R>,
+) -> Result<(
+    Option<ReadRef<R>>,
+    Arc<Vec<ReadRef<PlainIssue>>>,
+    Arc<Vec<ReadRef<PlainDiagnostic>>>,
+)> {
+    let result = source.strongly_consistent().await;
+    let issues = get_issues(source).await?;
+    let diagnostics = get_diagnostics(source).await?;
+
+    let result = if issues.iter().any(|i| i.severity <= IssueSeverity::Error) {
+        result.ok()
+    } else {
+        Some(result?)
+    };
```

Comment:
When there isn't a panic, we could just return the result.

```suggestion
    let result = if !result.is_ok() && issues.iter().any(|i| i.severity <= IssueSeverity::Error) {
        result.ok()
    } else {
        Some(result?)
    };
```